File: demo/src/main/java/com/example/demo/dto/Card.java
package com.example.demo.dto;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "cards") // Se guardarán en la colección "cards" de MongoDB
public class Card {

    @Id
    private String _id; // MongoDB usa _id

    private String name;
    private String type;
    private String description;
    private int cost;
    private int attack;
    private int defense; // Agregamos defensa
    private int health;
    private String faction;
    private String imageUrl; // El nombre del recurso (ej: "espadachin_solar")

    public Card() {}

    public Card(String name, String type, String description, int cost, int attack, int defense, int health, String faction, String imageUrl) {
        this.name = name;
        this.type = type;
        this.description = description;
        this.cost = cost;
        this.attack = attack;
        this.defense = defense;
        this.health = health;
        this.faction = faction;
        this.imageUrl = imageUrl;
    }
}

File: app/src/main/java/com/example/actividad2/data/model/Card.kt
package com.example.actividad2.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "cards")
data class Card(
    @PrimaryKey
    val id: String,
    val name: String,
    val cost: Int,
    val attack: Int,
    val health: Int,
    val type: String,
    val faction: String,
    val description: String,
    val imageResId: Int
)